test/tests/rackhd20/test_rackhd20_api_notifications.py

```diff
@@ -0,0 +1,267 @@
+'''
+Copyright 2017 Dell Inc. or its subsidiaries.  All Rights Reserved.
+Author(s):
+'''
+
+from time import sleep
+import Queue
+import flogging
+import logging
+import pika
+import unittest
+import threading
+import fit_common
+from nose.plugins.attrib import attr
+logs = flogging.get_loggers()
+amqp_queue = Queue.Queue(maxsize=0)
+from pymongo import MongoClient
+from bson.objectid import ObjectId
+
+IP = None
+amqp_port = None
+
+
+class AmqpWorker(threading.Thread):
+    def __init__(self, exchange_name, topic_routing_key, external_callback, timeout=10):
+        global amqp_port
+        threading.Thread.__init__(self)
+        pika_logger = logging.getLogger('pika')
+        if fit_common.VERBOSITY >= 8:
+            pika_logger.setLevel(logging.DEBUG)
+        elif fit_common.VERBOSITY >= 4:
+            pika_logger.setLevel(logging.WARNING)
+        else:
+            pika_logger.setLevel(logging.ERROR)
+        amqp_port = fit_common.fitports()['amqp_ssl']
+
+        if fit_common.API_PORT == "9090":
+            MONGO_PORT = 37017
+        else:
+            MONGO_PORT = 27017
+            logs.debug_1("{0}".format(MONGO_PORT))
+
+        self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=fit_common.fitargs()["rackhd_host"],
+                                                                            port=amqp_port))
+        self.channel = self.connection.channel()
+        result = self.channel.queue_declare(exclusive=True)
+        queue_name = result.method.queue
+        self.channel.queue_bind(exchange=exchange_name, queue=queue_name, routing_key=topic_routing_key)
+        self.channel.basic_consume(external_callback, queue=queue_name)
+        self.connection.add_timeout(timeout, self.dispose)
+
+    def dispose(self):
+        logs.debug_3('Pika connection timeout')
+        if self.connection.is_closed is False:
+            self.channel.stop_consuming()
+            logs.debug_1("Attempting to close the Pika connection")
+            self.thread_stop = True
+
+    def run(self):
+        logs.debug_7('start consuming')
+        self.channel.start_consuming()
+
+
+@attr(all=True, regression=False, smoke=True)
+class test_alert_notification(unittest.TestCase):
+    CLIENT = None
+    MONGO_PORT = None
+    CATALOGS_COUNT = None
+    NODES_COUNT = None
+    OBMS_COUNT = None
+
+    @classmethod
+    def setUpClass(self):
+        if fit_common.API_PORT == "9090":
```

Comment:
We shouldn't be selecting ports based on API_PORT deployment.  Running tests outside of vagrant (off-box) should be using the stack configuration 'vagrant-remote'.  Otherwise the stack configuration should be 'vagrant'.  At this point in the test code all port assignments that are based on deployment should be selected.   For example:

I think it should look something like:
MONGO_PORT = fit_common.fitports().get('mongo_port')
IP = fit_common.fit_common.fitcfg().get('rackhd_host')

No selection based on if fit_common.API_PORT == "9090"

mongo_port would be defaulted in install_config.  It could be overridden then in the stack_config.json.

The vagrant_remote needs to be populated accordingly.